sound/soc/sof/trace.c

```diff
@@ -43,6 +43,12 @@ static size_t sof_wait_trace_avail(struct snd_sof_dev *sdev,
 	if (ret)
 		return ret;
 
+	if (!sdev->dtrace_is_enabled && sdev->dtrace_eof) {
+		sdev->dtrace_eof = false;
```

Comment:
once this dtrace_eof is false, when does it become true again, if ever?

Reply:
@plbossart The eof (draining in latest) is set true whenever FW tracing is disabled. This happens e.g. every time runtime PM kicks in and suspends the DSP. We (might) have traces in DDR that have not bee read by user-space. We want to give user-space a chance to read these traces out, but not get confused when DSP resumes and starts writing from 0 again.